sdk/anomalydetector/azure-ai-anomalydetector/src/samples/java/com/azure/ai/anomalydetector/MultivariateSample.java

```diff
@@ -85,105 +91,103 @@ private static AnomalyDetectorClient getClient(String endpoint, String key) {
     }
 
     private static void getModelList(AnomalyDetectorClient client, Integer skip, Integer top) {
-        RequestOptions requestOptions = new RequestOptions()
-            .addQueryParam("skip", skip.toString())
-            .addQueryParam("top", top.toString());
-        PagedIterable<BinaryData> response = client.listMultivariateModels(requestOptions);
-
-        Iterator<PagedResponse<BinaryData>> ite = response.iterableByPage().iterator();
+        PagedIterable<Model> response = client.listMultivariateModels(skip, top);
+        Iterator<PagedResponse<Model>> ite = response.iterableByPage().iterator();
         int i = 1;
         while (ite.hasNext()) {
-            PagedResponse<BinaryData> items = ite.next();
+            PagedResponse<Model> items = ite.next();
             System.out.println("The result in the page " + i);
             i++;
-            for (BinaryData item : items.getValue()) {
-                JsonObject jsonObject = Json.createReader(new StringReader(item.toString())).readObject();
-                System.out.println("\t" + jsonObject.getString("modelId"));
+            for (Model item : items.getValue()) {
+                System.out.println("\t" + item.getModelId());
             }
         }
     }
 
-    private static Response<BinaryData> getLastDetectResult(AnomalyDetectorClient client, BinaryData body, UUID modelId) {
-        RequestOptions requestOptions = new RequestOptions();
-        Response<BinaryData> response = client.detectMultivariateLastAnomalyWithResponse(modelId.toString(), body, requestOptions);
-
-        return response;
+    private static LastDetectionResult getLastDetectResult(AnomalyDetectorClient client, LastDetectionRequest body, UUID modelId) {
+        LastDetectionResult res = client.detectMultivariateLastAnomaly(modelId.toString(), body);
+        return res;
     }
 
-    public static void run(BinaryData trainBody, BinaryData beginInferBody) throws Exception {
+    public static void run(String datasource, DataSchema dataSchema) throws Exception {
         String endpoint = "<anomaly-detector-resource-endpoint>";
         String key = "<anomaly-detector-resource-key>";
 
         // Get multivariate client
         AnomalyDetectorClient client = getClient(endpoint, key);
 
+        // set training request
+        OffsetDateTime startTime = OffsetDateTime.of(2021, 1, 2, 0, 0, 0, 0, ZoneOffset.UTC);
+        OffsetDateTime endTime = OffsetDateTime.of(2021, 1, 2, 5, 0, 0, 0, ZoneOffset.UTC);
+        ModelInfo trainRequest = new ModelInfo(datasource, startTime, endTime);
+        trainRequest.setSlidingWindow(200);
+        AlignPolicy alignPolicy = new AlignPolicy();
+        alignPolicy.setAlignMode(AlignMode.OUTER);
+        alignPolicy.setFillNAMethod(FillNAMethod.LINEAR);
+        alignPolicy.setPaddingValue(0.0);
+        trainRequest.setAlignPolicy(alignPolicy);
+        trainRequest.setDataSchema(dataSchema);
+        trainRequest.setDisplayName("SampleRequest");
```

Comment:
You can use the fluent style
e.g.
```java
trainRequest.setSlidingWindow(200)
    .setAlignPolicy(new AlignPolicy()
        .setAlignMode(AlignMode.OUTER)...)....
```

PS: is it correct that all properties in AlignPolicy be optional?

Reply:
updated.
Yes, AlignPolicy is optional.